docs/transfer_guard.rst

```diff
@@ -0,0 +1,71 @@
+Transfer guard
+==============
+
+JAX may transfer data between the host and devices and between devices during
+type conversion and input sharding. To log or disallow any unintended
+transfers, the user may configure a JAX transfer guard.
+
+JAX transfer guards distinguish between two types of transfers:
+
+* Explicit transfers: ``jax.device_put*()`` and ``jax.device_get()`` calls.
+* Implicit transfers: Other transfers (e.g., printing a ``DeviceArray``).
+
+A transfer guard can take an action based on its guard level:
+
+* ``"allow"``: Silently allow all transfers (default).
+* ``"log"``: Log and allow implicit transfers. Silently allow explicit
+  transfers.
+* ``"disallow"``: Disallow implicit transfers. Silently allow explicit
+  transfers.
+* ``"log_explicit"``: Log and allow all transfers.
+* ``"disallow_explicit"``: Disallow all transfers.
+
+The transfer guards use the standard JAX configuration system:
+
+* A ``--jax_transfer_guard=GUARD_LEVEL`` command-line flag and
+  ``jax.config.update("jax_transfer_guard", GUARD_LEVEL)`` will set the global
+  option.
+* A ``with jax.transfer_guard(GUARD_LEVEL): ...`` context manager will set the
+  thread-local option within the scope of the context manager.
+
+Note that similar to other JAX configuration options, a newly spawned thread
+will use the global option instead of any active thread-local option of the
+scope where the thread was spawned.
+
+The transfer guards also allow fine-control the transfer guard level of
+individual transfer directions. The flag and context manager name is suffixed
```

Comment:
For the first sentence here, how about the following instead?

"The transfer guards can also be applied more selectively, based on the direction of transfer."

Reply:
Thanks, updated.